plugins/src/android/OptimoveSDKPlugin.java

```diff
@@ -48,11 +86,142 @@ public boolean execute(String action, JSONArray args, CallbackContext callbackCo
+            return true;
+
+        case IN_APP_GET_INBOX_ITEMS:
+            cordova.getThreadPool().execute(() -> OptimoveSDKPlugin.this.inAppGetInboxItems(callbackContext));
+            return true;
+
+        case IN_APP_MARK_ALL_INBOX_ITEMS_AS_READ:
+            cordova.getThreadPool().execute(() -> OptimoveSDKPlugin.this.inAppMarkAllInboxItemsAsRead(callbackContext));
+            return true;
+
+        case IN_APP_MARK_AS_READ:
+            cordova.getThreadPool().execute(() -> OptimoveSDKPlugin.this.inAppMarkAsRead(args, callbackContext));
+            return true;
+
+        case IN_APP_GET_INBOX_SUMMARY:
+            this.inAppGetInboxSummary(callbackContext);
+            return true;
+
+        case IN_APP_PRESENT_INBOX_MESSAGE:
+            cordova.getThreadPool()
+                    .execute(() -> OptimoveSDKPlugin.this.inAppPresentInboxMessage(args, callbackContext));
+            return true;
+
+        case IN_APP_DELETE_INBOX_MESSAGE:
+            cordova.getThreadPool()
+                    .execute(() -> OptimoveSDKPlugin.this.inAppDeleteMessageFromInbox(args, callbackContext));
+            return true;
         }
 
         return false;
     }
 
+    private void inAppDeleteMessageFromInbox(JSONArray args, CallbackContext callbackContext) {
+        int messageId = args.optInt(0, -1);
+
+        if (messageId == -1) {
+            callbackContext.error("Message not found or not available");
+            return;
+        }
+
+        List<InAppInboxItem> items = OptimoveInApp.getInstance().getInboxItems();
+        for (InAppInboxItem item : items) {
+            if (item.getId() == messageId) {
+                boolean result = OptimoveInApp.getInstance().deleteMessageFromInbox(item);
+
+                if (result) {
+                    callbackContext.success();
+                    return;
+                }
+
+                break;
+            }
+        }
+
+        callbackContext.error("Message not found or not available");
+    }
+
+    private void inAppPresentInboxMessage(JSONArray args, CallbackContext callbackContext) {
+        int messageId = args.optInt(0, -1);
+
+        if (messageId == -1) {
+            callbackContext.error("Message not found or not available");
+            return;
+        }
+
+        List<InAppInboxItem> items = OptimoveInApp.getInstance().getInboxItems();
+        for (InAppInboxItem item : items) {
+            if (item.getId() == messageId) {
+                OptimoveInApp.InboxMessagePresentationResult result = OptimoveInApp.getInstance()
+                        .presentInboxMessage(item);
+
+                if (result == OptimoveInApp.InboxMessagePresentationResult.PRESENTED) {
+                    callbackContext.success();
+                    return;
+                } else {
+                    break;
+                }
+            }
+        }
+
+        callbackContext.error("Message not found or not available");
```

Comment:
Do we want to consider passing back the in app presentation result value here?

Reply:
(if available)

Reply:
I think in the discussion yesterday we decided to keep it the way it is